pkg/kudoctl/packages/verifier/template/verify_parameters.go

```diff
@@ -43,13 +49,85 @@ func (ParametersVerifier) Verify(pf *packages.Files) verifier.Result {
 
 func immutableParams(pf *packages.Files) verifier.Result {
 	res := verifier.NewResult()
-	for _, value := range pf.Params.Parameters {
-		if value.IsImmutable() {
-			if !value.HasDefault() && !value.IsRequired() {
-				res.AddParamError(value.Name, "is immutable but is not marked as required or has a default value")
+	for _, p := range pf.Params.Parameters {
+		if p.IsImmutable() {
+			if !p.HasDefault() && !p.IsRequired() {
+				res.AddParamError(p.Name, "is immutable but is not marked as required or has a default value")
+			}
+		}
+	}
+	return res
+}
+
+func paramDefaults(pf *packages.Files) verifier.Result {
+	res := verifier.NewResult()
+	for _, p := range pf.Params.Parameters {
+		if p.HasDefault() {
+			if err := p.ValidateDefault(); err != nil {
+				res.AddErrors(err.Error())
+			}
+		}
+	}
+	return res
+}
+
+func enumParams(pf *packages.Files) verifier.Result {
+	res := verifier.NewResult()
+	for _, p := range pf.Params.Parameters {
+
+		if p.IsEnum() {
+			if len(p.EnumValues()) == 0 {
+				res.AddParamError(p.Name, "is an enum but has no allowed values")
+				continue
+			}
+			for _, enumVal := range p.EnumValues() {
+
+				if err := kudoapi.ValidateParameterValueForType(p.Type, enumVal); err != nil {
```

Comment:
why is this enum validation not part of some kind of `p.Validate` as well? 🤔 then `ValidateParameterValueForType` could have been made private

Reply:
Well, one reason is that here in this function we have a `package.Parameter` and not a `kudoapi.Parameter` - This caused quite a bit of headache while organising this - both types are distinct but share quite a bit of common functionality. 
This is one of the cases where I'd really like to have generics ;)